File: ASP.NET Web Deployment using Visual Studio/[C#]-ASP.NET Web Deployment using Visual Studio/ContosoUniversity-End/ContosoUniversity.DAL/Student.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ContosoUniversity.DAL
{
    public class Student : Person
    {
        public Student()
        {
            Enrollments = new List<Enrollment>();
        }

        [Required(ErrorMessage = "Enrollment date is required.")]
        [DisplayFormat(DataFormatString = "{0:d}", ApplyFormatInEditMode = true)]
        [Display(Name = "Enrollment Date")]
        public DateTime? EnrollmentDate { get; set; }

        public virtual ICollection<Enrollment> Enrollments { get; set; }
    }
}

File: ASP.NET Web Forms Application Using Entity Framework 4.0 Database First/[VB]-ASP.NET Web Forms Application Using Entity Framework 4.0 Database First/VB/ContosoUniversity/DAL/Student.vb
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations

<MetadataType(GetType(StudentMetadata))>
    Partial Public Class Student
End Class

Public Class StudentMetadata
    <DisplayFormat(DataFormatString:="{0:d}", ApplyFormatInEditMode:=True)>
    Public Property EnrollmentDate As Date

    <StringLength(25, ErrorMessage:="First name must be 25 characters or less in length."), Required(ErrorMessage:="First name is required.")>
    Public Property FirstMidName As String

    <StringLength(25, ErrorMessage:="Last name must be 25 characters or less in length."), Required(ErrorMessage:="Last name is required.")>
    Public Property LastName As String
End Class
